src/strategies/tomb-finance/index.ts

```diff
@@ -0,0 +1,219 @@
+export const version = '0.1.0';
+
+const MASONRY_ADDRESS = '0x8764DE60236C5843D9faEB1B638fbCE962773B67';
+const TSHARE_TOKEN_ADDRESS = '0x4cdF39285D7Ca8eB3f090fDA0C069ba5F4145B37';
+const TSHARE_LP_TOKEN_ADDRESS = '0x4733bc45eF91cF7CcEcaeeDb794727075fB209F2';
+const CEMETERY_ADDRESS = '0xcc0a87F7e7c693042a9Cc703661F5060c80ACb43';
+
+const PRECISION = BigNumber.from(10).pow(18);
+
+const abi = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    name: 'balanceOf',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  },
+  {
+    inputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      },
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    name: 'userInfo',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256'
+      },
+      {
+        internalType: 'uint256',
+        name: 'rewardDebt',
+        type: 'uint256'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  },
+  {
+    constant: true,
+    inputs: [],
+    name: 'totalSupply',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  },
+  {
+    constant: true,
+    inputs: [],
+    name: 'strategy',
+    outputs: [
+      {
+        internalType: 'address',
+        name: '',
+        type: 'address'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  let multi = new Multicaller(network, provider, abi, { blockTag });
+
+  options.vaultTokens.forEach((token: any) => {
+    multi.call(`vaultStrategies.${token.address}`, token.address, 'strategy');
+  });
+
+  const vaultStrategiesResult = await multi.execute();
+
+  multi = new Multicaller(network, provider, abi, { blockTag });
+
+  multi.call(`lp.tshareBalance`, TSHARE_TOKEN_ADDRESS, 'balanceOf', [
+    TSHARE_LP_TOKEN_ADDRESS
+  ]);
+  multi.call(`lp.totalSupply`, TSHARE_LP_TOKEN_ADDRESS, 'totalSupply');
+  options.vaultTokens.forEach((token: any) => {
+    multi.call(
+      `vaultTokens.totalSupply.${token.address}`,
+      token.address,
+      'totalSupply',
+      []
+    );
+    multi.call(
+      `vaultTokens.lpBalance.${token.address}`,
+      CEMETERY_ADDRESS,
+      'userInfo',
+      ['1', vaultStrategiesResult.vaultStrategies[token.address]]
+    );
+  });
+
+  addresses.forEach((address: any) => {
+    multi.call(`tshare.${address}`, TSHARE_TOKEN_ADDRESS, 'balanceOf', [
```

Comment:
Multicalls for each address won't scale, is there any other alternate way to calculate the score?